gufe/protocols/protocol.py

```diff
@@ -270,6 +268,78 @@ def create(
             extends_key=extends.key if extends is not None else None,
         )
 
+    @abc.abstractmethod
+    def _validate(
+        self,
+        *,
+        stateA: ChemicalSystem,
+        stateB: ChemicalSystem,
+        mapping: Optional[Union[ComponentMapping, list[ComponentMapping]]],
+        extends: Optional[ProtocolDAGResult] = None,
+    ):
+        """Method to override in custom `Protocol` subclasses.
+
+        This method should take two `ChemicalSystem`s, and optionally
+        one or more `ComponentMapping` objects and `ProtocolDAGResult`
+        objects used for extensions, and validate them, raising a
+        `ValueError` where validation is not successful.
```

Comment:
I think we should start pointing people to our new validation errors https://github.com/OpenFreeEnergy/gufe/blob/ab2ca232365e78c5a252bbdfdf4bc03c6277fee6/gufe/protocols/errors.py#L10

Reply:
Spoken like the author of such an exception! Done!